File: core/src/com/mygdx/game/handler/StartWalkHandler.java
package com.mygdx.game.handler;

public class StartWalkHandler extends BaseHandler {

    @Override
    public BaseHandler.HandlerEntity doHandle(BaseHandler.HandlerEntity s, HandlerChain chain) {

        System.out.println("开始了");
//        String dotask = dotask();
        System.out.println("收到回复");
        return chain.process(s);
    }

    public String dotask() {
        ResultReporter<String> reporter = new ResultReporter<>();
        ReportUtils.deley(1, reporter, "开始吧");
        return reporter.waitReport();

    }
}

File: core/src/main/kotlin/com/rich/diy/game/handler/StartWalkHandler.kt
package com.rich.diy.game.handler

class StartWalkHandler : BaseHandler() {
    override fun doHandle(s: HandlerEntity, chain: HandlerChain): HandlerEntity? {
        if (s.player!!.equals("dan")) {
//            s.setParcelData("StopWayHandler");
        }
        println("开始了")
        //        String dotask = dotask();
        println("收到回复")
        return chain.process(s)
    }

    fun dotask(): String? {
        val reporter = ResultReporter<String>()
        //        Main.task1(reporter);
        return reporter.waitReport()
    }
}